pkg/eventqueue/eventqueue_test.go

```diff
@@ -213,3 +213,27 @@ func (s *EventQueueSuite) TestEnqueueTwice(c *C) {
 	q.Stop()
 	q.WaitToBeDrained()
 }
+
+func (s *EventQueueSuite) TestForcefulDraining(c *C) {
+	// This will test enqueuing an event when the queue was never run and was
+	// stopped and drained. The behavior expected is that the event will
+	// successfully be enqueued (channel returned is non-nil & no error), and
+	// after the event is stopped and drained, the returned channel will
+	// unblock.
+
+	q := NewEventQueue()
+
+	ev := NewEvent(&DummyEvent{})
+	res, err := q.Enqueue(ev)
+	c.Assert(res, Not(IsNil))
+	c.Assert(err, IsNil)
+
+	q.Stop()
+	q.WaitToBeDrained()
+
+	select {
+	case <-res:
+	case <-time.After(5 * time.Second):
```

Comment:
is 5s standard for timeouts in cilium test cases? seems pretty low for a busy CI system (read: possible test flake generator)

Reply:
I don't think we have a standard timeout in unit tests. Just doing a quick, non-exhaustive grep over the codebase gives:

```shell
❯ rg -i "testtimeout .+ =.+ " --glob '!vendor' -c
pkg/completion/completion_test.go:1
pkg/envoy/xds/server_e2e_test.go:1
proxylib/npds/client_test.go:1
```

It seems like we define a specific timeout (if necessary) for code that is asynchronous, depending on the test itself. In this case though for the eventqueue, the `res` channel should be ready to receive almost immediately because the event is a dummy event. While yes it's possible that on a very contended system, 5 seconds for Go to schedule the `select` might be too little, but the likelihood of that is extremely small. This test has been around for probably over a year now, so it seems to be OK.